utils/svg2png/svg2png.cpp

```diff
@@ -71,17 +73,34 @@ struct main_marker_visitor
         double opacity = 1;
         double w, h;
         std::tie(w, h) = marker.dimensions();
+
+
+        double svg_width = w * scale_factor_ + 0.5;
+        double svg_height = h * scale_factor_ + 0.5;
+
+        int output_width = static_cast<int>(svg_width);
+        int output_height = static_cast<int>(svg_height);
         if (verbose_)
         {
-            std::clog << "found width of '" << w << "' and height of '" << h << "'\n";
+            std::clog << "Found width of '" << w << "' and height of '" << h << "'\n";
+            auto b = marker.bounding_box();
+            std::clog << "SVG BBOX:" << b << std::endl;
+            std::clog << "Output image dimensions:[" << output_width << "," << output_height << "]" << std::endl;
         }
-        mapnik::image_rgba8 im(static_cast<int>(w + 0.5), static_cast<int>(h + 0.5));
+        mapnik::image_rgba8 im(output_width, output_height);
+        im.set(0xffffffff);
         agg::rendering_buffer buf(im.bytes(), im.width(), im.height(), im.row_size());
         pixfmt pixf(buf);
         renderer_base renb(pixf);
 
-        mapnik::box2d<double> const& bbox = {0, 0, w, h};
-        agg::trans_affine mtx = {};
+        mapnik::box2d<double> const& bbox = {0, 0, svg_width, svg_height};
+        // center the svg marker on '0,0'
+        agg::trans_affine mtx = agg::trans_affine_translation(-0.5 * w, -0.5 * h);
+        // Scale the image
+        mtx.scale(scale_factor_);
+        // render the marker at the center of the marker box
+        mtx.translate(0.5 * svg_width, 0.5 * svg_height);
```

Comment:
If I'm reading this correctly, this matrix results in scaling and translation by 0.25px: it shifts by half width, then scales, then shifts back by half scaled width. If svg_width didn't have 0.5 added to it, the matrix would be equivalent to just scaling.

Reply:
I copied it from https://github.com/mapnik/node-mapnik/blob/master/src/mapnik_image.cpp#L2833-L2838  as I wanted exactly the same behaviour to start with. But, you're right it seems redundant. Great to have your eyes looking at this, cheers!

Reply:
I created an issue to look into this #4116